ext/hosts

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python2
+#
+# Build hosts-like file with data from external inventory (DNS, droplets API)
+
+import json
+import socket
+from  subprocess import check_output
+import xml.etree.ElementTree as ElementTree
+
+dns = {}
+gh = {}
+ptr = {}
+state = {}
+
+ns = {'NC': 'http://api.namecheap.com/xml.response'}
+for domain in ('openobservatory.org', 'ooni.io', 'ooni.nu'):
+    tree = ElementTree.parse(domain + '.xml')
+    for el in tree.findall('./NC:CommandResponse/NC:DomainDNSGetHostsResult/NC:host', ns):
+        if el.get('Type') == 'A':
+            dns.setdefault(el.get('Address'), set()).add(el.get('Name') + '.' + domain)
+
+droplets = json.load(open('gh-droplets.json'))
+for vm in droplets['droplets']:
+    for net in vm['networks']['v4']: # no IPv6 support there yet
+        gh[net['ip_address']] = vm['name']
+
+all_ips = dns.viewkeys() | gh.viewkeys()
+
+nmap = check_output(['sudo', 'nmap', '-v', '-sn', '-oX', '-'] + list(all_ips))
```

Comment:
That's a script to be run at controller host that is your/mine PC (as we agreed to avoid separate controller jumphost with all private keys) that likely has nmap installed :)